### PR TITLE
Typing Part 1: Fix Makefile and add mypy

### PR DESCRIPTION
**Pull request by herostrat**
This commit fixes the make_tests Makefile entry and adds support for mypy.

Also: Please consider making the dev branch the base for integration (default)
See https://github.com/blakeblackshear/frigate/discussions/3081

### CODE REVIEW

File: mypy.ini [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[mypy]
+python_version = 3.7
```
Comment (blakeblackshear) [inline mypy.ini:2]: I haven't used mypy before. Is it typical to use 3.7 for type checking even when the version you are running is 3.9?

Comment (herostrat) [reply mypy.ini:2]: No, you this was just from experiments to check multiple different python version.
Will change to 3.9

**Comment by herostrat** [comment]
I think I got it to work.

The second commit fixes:
./opt/frigate/frigate/util.py:37: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
  if isinstance(v1, dict) and isinstance(v2, collections.Mapping):

**Comment by herostrat** [comment]
This should be ready for review.
As I stated I am a docker noob, please be wary!

**Pull request merged**